source/develop/release-management/features/sla/hosted-engine-network-check.html.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Alternative Network Check for Self Hosted Engine
+category: feature
+authors: dholler
+feature_name: Alternative Network Check for Self Hosted Engine
+feature_modules: ovirt-hosted-engine-ha, cockpit-ovirt, ovirt-ansible-hosted-engine-setup, ovirt-hosted-engine-setup
+feature_status: Released
+---
+
+# Alternative Network Check for Self Hosted Engine
+
+## Summary
+
+Currently ovirt-hosted-engine-ha, the agent running on potential hosts to host
+the Self Hosted Engine VM, checks his network liveliness by pinging the
+gateway.
+This feature adds alternative ways to check network liveliness, to enable
+Self Hosted Engine in networks which does not allow pinging the gateway.
+
+The progress of the implementation and testing of this feature is tracked on
+[Bug 1659052 - [RFE] Consider alternative to ICMP ping for hosted engine network test](https://bugzilla.redhat.com/1659052)
+
+## Owner
+
+*   Name: [Dominik Holler](https://github.com/dominikholler)
+
+*   Email: <dholler@redhat.com>
+
+
+## Detailed Description
+
+Additional ways to check the network liveliness to be added are DNS, TCP and
+none.
+
+DNS means that an NS query for "." (the root) will be performed, if this
+succeeds, network is considered as alive.
+
+TCP means, that a TCP connection to a given host and port combination will be
+created. If this succeeds, network is considered as alive.
+
+None means, that the network is always considered alive.
```

Comment:
Is that ever the case, I mean is this safe to be provided as a choice?

Reply:
@tiraboschi can you please jump in here?

Reply:
sanlock (volume lease) is used to protect the engine VM disk at storage level: we are sure that two copies of the same VM acting on the same disk could not be started even if the two hosts cannot communicate over the network so from this point of view it's absolutely safe.

On the other side, if the network on the host is not properly working (various possible issues here with different effects) nothing will be going to restart the engine VM over a different host and so we are not 100% sure that the user can really reach the engine VM from his laptop.